sql/core/src/test/scala/org/apache/spark/sql/CollationStringExpressionsSuite.scala

```diff
@@ -89,6 +89,45 @@ class CollationStringExpressionsSuite
     testRepeat("UNICODE_CI", 3, "abc", 2)
   }
 
+  test("UPPER check output type on collated string") {
```

Comment:
Maybe it's more readable to combine multiple simple checks for expressions like this into one test function like [here](https://github.com/apache/spark/pull/45933/files#diff-ab481fd0c9704381d8535339727e08673da432080e47be49ab002411adef2afa). In this case we could also combine `testUpper`, `testLower`, `testInitCap` into one function.

Reply:
Yeah, agreed. Will refactor a bit.